src/common/java/net/minecraftforge/gradle/common/util/DownloadUtils.java

```diff
@@ -193,9 +191,8 @@ public static File downloadWithCache(URL url, File target, boolean changing, boo
         } catch (IOException e) {
             //Eat it, some repos don't have a simple checksum.
         }
-        if (expected == null && bypassLocal) return null; // Ignore local file if the remote doesn't have it.
-        if (expected == null && target.exists()) return target; //Assume we're good cuz they didn't have a MD5 on the server.
-        if (expected != null && expected.equals(actual)) return target;
+        if (expected == null && bypassLocal) return null; // Ignore local file if the remote doesn't have a MD5 checksum.
+        if (expected != null && expected.equals(actual)) return target; // Require MD5 checksum to skip the download as it will be calculated after a successful download.
```

Comment:
So now that this is only possible to get via the md5 file being missing and us having one that matches the server, we should update the local md5 file to the correct hash.